File: src/main/java/fr/ynerant/leveleditor/game/NullTower.java
package fr.ynerant.leveleditor.game;

import java.util.ArrayList;
import java.util.Collection;

public class NullTower extends Tower {
    public NullTower(int x, int y) {
        super(x, y);
    }

    @Override
    public int getDamagePerShot() {
        return Integer.MAX_VALUE;
    }

    @Override
    public long getPeriod() {
        return 1;
    }

    @Override
    public int getPrice() {
        return 0;
    }

    @Override
    Collection<Mob> _filterDetectedMobs(Collection<Mob> mobs) {
        return new ArrayList<>();
    }
}

File: src/main/scala/fr/ynerant/leveleditor/game/towers/NullTower.scala
package fr.ynerant.leveleditor.game.towers

import fr.ynerant.leveleditor.game.mobs.Mob
import java.util


class NullTower(override val x: Int, override val y: Int) extends Tower(x, y) {
	override def getName = "nulltower"

	override def getDamagePerShot: Int = Integer.MAX_VALUE

	override def getPeriod = 1

	override def getPrice = 5

	override private[towers] def _filterDetectedMobs(mobs: util.Collection[Mob]) = new util.ArrayList[Mob]
}
